nixos/modules/misc/ids.nix

```diff
@@ -356,6 +356,7 @@ in
       rstudio-server = 324;
       localtimed = 325;
       automatic-timezoned = 326;
+      certspotter = 327;
```

Comment:
Is this needed? I think DynamicUser will work for this service

Reply:
no, it supports arbitrary shell hooks, so we need an actual user

Reply:
Unresolving this. I believe `users.users.certspotter.isSystemUser = true` should cover your needs.

Reply:
(Note that using `StateDirectory` causes UID stability across e.g. disabling and re-enabling of the service in NixOS configuration to not be a source of issues, due to systemd updating permissions of the directory and its contents. Users of this module who need to have e.g. directories for use by hooks can use `systemd.tmpfiles` to get the same result for those directories.)

Reply:
From a helpful chap (dali99) on matrix: 

> uid and gids are tracked in /var/lib/nixos/{u,g}id-map and should be stable for the same installation

Reply:
I've switched to `isSystemUser`, but just so we're on the same page, where is adding a uid/gid to `ids.nix` appropriate then?

Reply:
I think next to never. The cases I can come up with:
 - when you need the UID to match across machines,
 - when the UID needs to match a constant (e.g. in a package),
 - to deal with some weird edge cases of the uid selection system.

(Note that the `[ug]id-map` files contain [UG]IDs for groups that don't exist anymore, too, so if they get recreated under the same name they get the same UID back. I didn't know this when I was writing my previous comment.)

Reply:
ok, I don't think the existing uid/gid allocations there are all necessary then, it's a little confusing

Reply:
we're migrating away from it, simply because 1000 uids and gids is not enough. An if you do need to lock the ids, say if the state is hosted on a NAS over nfs, then you are free to do so in your own nixos config